libmscore/figuredbass.h

```diff
@@ -277,6 +278,8 @@ class FiguredBass : public Text {
       static bool       fontData(int nIdx, QString *pFamily, QString *pDisplayName,
                               qreal * pSize, qreal * pLineHeight);
 
+      virtual std::vector<FiguredBassItem*> getFiguredBaseItems() { return items; };
+
```

Comment:
indeed this is the big question.  As I understand, the copy constructor already returns a copy of the vector, so the outside world cannot modify the vector.

In the mean time, I believe I’m only using getFiguredBassItems from within the figuredbass.cpp file.

I may need to change that because in writing the code within this file, I am violating other code separation issues.
E.g., I have a #include “mscore/scoreview.h“ at the top which is apparently forbidden.



> On 24 May 2015, at 05:11, Maurizio M. Gavioli <notifications@github.com> wrote:
> 
> In libmscore/figuredbass.h <https://github.com/musescore/MuseScore/pull/2015#discussion_r30949788>:
> 
> > @@ -277,6 +278,8 @@ class FiguredBass : public Text {
> >        static bool       fontData(int nIdx, QString *pFamily, QString *pDisplayName,
> >                                qreal * pSize, qreal * pLineHeight);
> >  
> > +      virtual std::vector<FiguredBassItem*> getFiguredBaseItems() { return items; };
> > +
> Please not public. Or at least const std::vector<const FiguredBassItem*> (not sure I got the syntax right, but the gol is that neither the vector nor each of the items should be modifiable from outside).
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/musescore/MuseScore/pull/2015/files#r30949788>.
>